vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1000,7 +1000,16 @@ def execute_model(
                 num_scheduled_tokens)
         else:
             # Eager mode.
-            num_input_tokens = num_scheduled_tokens
+            # Pad tokens to multiple of tensor_parallel_size when
+            # enabled collective fusion for SP
+            tp_size = self.vllm_config.parallel_config.tensor_parallel_size
+            if self.vllm_config.compilation_config.pass_config. \
+                enable_sequence_parallelism and tp_size > 1:
+                import math
+                num_input_tokens = math.ceil(
+                    num_scheduled_tokens / tp_size) * tp_size
```

Comment:
nit: We have a `round_up` util for this (you can import it from `vllm/utils.py`)
```
num_input_tokens = round_up(num_scheduled_tokens, tp_size)
```